borg/helpers.py

```diff
@@ -98,6 +101,23 @@ def check_extension_modules():
         raise ExtensionModuleError
 
 
+SUPPORTED_REPO_FEATURES = frozenset()
+
+REPO_OPERATION_CHECK = "check"
+REPO_OPERATION_WRITE = "write"
+REPO_OPERATION_DELETE = "delete"
+
+def check_repository_compatibility(manifest, operation):
+    if manifest is None:
+        # various fake options can have manifest is None
+        return
+    if not operation in manifest.config:
+        return
+    requirements = manifest.config[operation]
+    if 'mandatory' in requirements and not set(requirements['mandatory']) <= SUPPORTED_REPO_FEATURES:
+        raise MandatoryFeatureUnsupported(set(requirements['mandatory']) - SUPPORTED_REPO_FEATURES)
```

Comment:
maybe always compute the unsupported_features set and raise if not empty? saves a bit of duplication maybe.